clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/job/KubectlJobExecutor.java

```diff
@@ -363,16 +368,53 @@ public KubernetesManifest get(KubernetesV2Credentials credentials, KubernetesKin
       }
     }
 
+    List<KubernetesManifest> manifestList = new ArrayList<>();
+
     if (status.getStdErr().contains("No resources found")) {
-      return new ArrayList<>();
+      return manifestList;
     }
 
+    InputStream resultStream = new ByteArrayInputStream(status.getStdOut().getBytes(StandardCharsets.UTF_8));
```

Comment:
Good point. With that in mind, I went looking at the JobExecutor code, the implementation is here: https://github.com/DataDog/clouddriver/blob/dd-master/clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/jobs/local/JobExecutorLocal.groovy#L114-L115

And it looks like this starts out as a `ByteArrayInputStream` anyway, then gets turned into a string here https://github.com/DataDog/clouddriver/blob/dd-master/clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/jobs/local/JobExecutorLocal.groovy#L123-L124

Does it make sense to adjust the JobExecutor to surface stderr / stdout as a `ByteArrayInputStream` vs `String` in that `JobStatus` object? Then let the caller decide what to do with the result? I don't see much usage of this, so I don't think it'd be a lot of work. Thoughts?

Reply:
if we were to go about changing the signature in the JobExecutor we would probably want to have it return something like guava's `ByteSource` or some similar riff on providing a safe handle to getting a new inputstream that you then close, rather than worry about ByteArrayInputStream vs String (this is predicated on the fact that ByteArrayInputStream not an open i/o stream somewhere, so much as an already created and fully populated array with code implementing InputStream around it)